arm/Microsoft.Synapse/workspaces/deploy.bicep

```
@@ -0,0 +1,293 @@
+//   scope: az.resourceGroup(encryptionKeyVaultResourceGroupName)
+// }
+
+// module workspace_cmk_rbac '.bicep/nested_cmkRbac.bicep' = if (encryptionActivateWorkspace) {
+//   name: '${workspace.name}-cmk-rbac'
+//   params: {
+//     workspaceIdentity: workspace.identity.principalId
+//     keyvaultName: encryptionKeyVaultName
+//     usesRbacAuthorization: keyVault.properties.enableRbacAuthorization
+//   }
+//   scope: az.resourceGroup(encryptionKeyVaultResourceGroupName)
+// }
+
+// // Workspace encryption - Activate Workspace
+// module workspace_cmk 'keys/deploy.bicep' = if (encryptionActivateWorkspace) {
+//   name: '${workspace.name}-cmk-activation'
+//   params: {
+//     isActiveCMK: true
+//     keyVaultUrl: keyVaultUrl
+//     name: encryptionKeyName
+//     workspaceName: workspace.name
+//   }
+//   dependsOn: [
+//     workspace_cmk_rbac
+//   ]
+// }
+
+// Resource Lock
+resource workspace_lock 'Microsoft.Authorization/locks@2016-09-01' = if (lock != 'NotSpecified') {
+  name: '${workspace.name}-${lock}-lock'
+  properties: {
+    level: lock
+    notes: (lock == 'CanNotDelete') ? 'Cannot delete resource or child resources.' : 'Cannot modify the resource or child resources.'
+  }
+  scope: workspace
+}
+
+// RBAC
+module workspace_rbac '.bicep/nested_rbac.bicep' = [for (roleAssignment, index) in roleAssignments: {
+  name: '${deployment().name}-rbac-${index}'
+  params: {
+    principalIds: roleAssignment.principalIds
+    roleDefinitionIdOrName: roleAssignment.roleDefinitionIdOrName
+    resourceId: workspace.id
+  }
+}]
+
+// Private Endpoints
+module workspace_privateEndpoints '.bicep/nested_privateEndpoint.bicep' = [for (privateEndpoint, index) in privateEndpoints: {
+  name: '${uniqueString(deployment().name, location)}-PrivateEndpoint-${index}'
+  params: {
+    privateEndpointResourceId: workspace.id
+    privateEndpointVnetLocation: reference(split(privateEndpoint.subnetResourceId, '/subnets/')[0], '2020-06-01', 'Full').location
+    privateEndpointObj: privateEndpoint
+    tags: tags
+  }
+}]
+
+// Diagnostics Settings
+resource workspace_diagnosticSettings 'Microsoft.Insights/diagnosticsettings@2021-05-01-preview' = if (!empty(diagnosticStorageAccountID) || !empty(workspaceID) || !empty(eventHubAuthorizationRuleID) || !empty(eventHubName)) {
+  name: '${workspace.name}-diagnosticSettings'
+  properties: {
+    storageAccountId: !empty(diagnosticStorageAccountID) ? diagnosticStorageAccountID : null
+    workspaceId: !empty(workspaceID) ? workspaceID : null
+    eventHubAuthorizationRuleId: !empty(eventHubAuthorizationRuleID) ? eventHubAuthorizationRuleID : null
+    eventHubName: !empty(eventHubName) ? eventHubName : null
+    logs: diagnosticsLogs
+  }
+  scope: workspace
+}
+
+@description('The resource ID of the deployed Synapse Workspace.')
+output workspaceResourceID string = workspace.id
+
+@description('The name of the deployed Synapse Workspace.')
+output workspaceName string = workspace.name
```

Latest convention (to align all modules with each other and the Azure Template reference)

```suggestion
output name string = workspace.name
```